src/utils/isSlotPopulated.ts

```diff
@@ -20,18 +20,23 @@
  *
  */
 
+import type { VNode, VNodeNormalizedChildren } from 'vue'
 import { Fragment, Comment, Text } from 'vue'
 
 /**
  * Checks whether a slot is populated
  *
- * @param {Array} vnodes The array of vnodes to check
+ * @param vnodes The array of vnodes to check
  */
-const isSlotPopulated = function(vnodes) {
-	return !!vnodes?.some(node => {
-		if (node.type === Comment) return false
-		if (node.type === Fragment && !isSlotPopulated(node.children)) return false
-		if (node.type === Text && !node.children.trim()) return false
+const isSlotPopulated = function(vnodes?: VNode[] | VNodeNormalizedChildren) {
+	return Array.isArray(vnodes) && vnodes.some(node => {
```

Comment:
Why do you prefer
```ts
Array.isArray(vnodes) && vnodes.some(...)
```
over
```js
!!vnodes?.some(...)
```
? Does it behave different in some cases?

Reply:
Because vnodes is not always an array, you call the function like this `isSlotPopulated(node.children)` so the the parameter type for that call is `VNodeNormalizedChildren` which can be `null, string, VNode[], RawSlots`.

So when you want to call `some` on that parameter it will cause a Typescript error:

> Property 'some' does not exist on type 'string | VNode<RendererNode, RendererElement, { [key: string]: any; }>[] | VNodeArrayChildren | RawSlots'.
> Property 'some' does not exist on type 'string'.ts(2339)

So Array.isArray simply asserts that the parameter is an Array, so the following code will assume that vnodes can only be of type `VNode[]`.

Reply:
Indeed, that's what I had in mind. But checking that's an array first seems like the better solution here. 👍